### ISSUE TITLE
Heapster v.1.1.0 fails against ES 2.3.2

### ISSUE THREAD

**Issue by Thermi** [kubernetes/heapster#1242]
Hello,

Heapster is not capable of connecting to ES nodes with version 2.3.2, despite communication succeeding. I think this is obviously a bug and should be fixed. If it is not, then please make the error message more descriptive.

Heapster logs:
```
 kubectl logs --container=heapster --namespace=kube-system heapster-v1.1.0-1183323529-pgv68 -f
I0804 23:54:24.496828       1 heapster.go:65] /heapster --source=kubernetes.summary_api:'' --metric_resolution=60s --sink=elasticsearch:?nodes=http://search-new-cluster-kzrbtkd6wjv7heombzbtd
wzsva.eu-west-1.es.amazonaws.com --sink=log --vmodule=*=4
I0804 23:54:24.496903       1 heapster.go:66] Heapster version 1.1.0
I0804 23:54:24.497092       1 configs.go:60] Using Kubernetes client with master "https://10.3.0.1:443" and version "v1"
I0804 23:54:24.497124       1 configs.go:61] Using kubelet port 10255
I0804 23:54:24.497704       1 reflector.go:202] Starting reflector *api.Node (1h0m0s) from k8s.io/heapster/metrics/sources/summary/summary.go:454
I0804 23:54:24.497735       1 elasticsearch.go:103] configing elasticsearch sink with ES's nodes - [http://search-new-cluster-kzrbtkd6wjv7heombzbtdwzsva.eu-west-1.es.amazonaws.com]
I0804 23:54:24.498293       1 reflector.go:282] Listing and watching *api.Node from k8s.io/heapster/metrics/sources/summary/summary.go:454
I0804 23:54:25.070488       1 meta.go:46] Calling Accessor on non-internal object: *api.NodeList
I0804 23:54:29.772034       1 driver.go:97] failed to config elasticsearch
E0804 23:54:29.772084       1 factory.go:74] Failed to create sink: failed to create ElasticSearch client: no Elasticsearch node available

```

PCAP with connection is attached.

[heapster-es-fail.pcap.zip](https://github.com/kubernetes/heapster/files/402913/heapster-es-fail.pcap.zip)

**Comment by AlmogBaku** [comment]
cc: @jamiehannaford @huangyuqi do you have any idea regarding this?

**Comment by Thermi** [comment]
Oh come on, at least write this somewhere in the manual or fix it in the code, god damnit.
Upstream already had a PR because of that problem and it says why it breaks, too: olivere/elastic#193

**Comment by jimmidyson** [comment]
> god damnit

I doubt He/She/It cares...

As you feel so strongly about it, how about sending in a PR that fixes it?

**Comment by AlmogBaku** [comment]
Hi, here is our findings by far:
1) AWS ES requires turning off sniffing - https://github.com/olivere/elastic/wiki/Using-with-AWS-Elasticsearch-Service
2) In order to use AWS ES you need to whitelist your cluster IPs or using the AWS Client with IAM permissions, which is not supported by the current implementation.
3) Heapster version 1.2 and above ONLY allow using AWS ES due to the updated olivere/elastic version and the support of disabling the sniffing:
```
--sink=elasticsearch:?nodes=http://my-cluster.eu-west-1.es.amazonaws.com&sniff=false
```

**Comment by AlmogBaku** [comment]
We still experiencing issues to push to the ES server:
```
2016-08-11T12:39:25.468847117Z W0811 12:39:25.468470       1 manager.go:119] Failed to push data to sink: ElasticSearch Sink
```

**Comment by AlmogBaku** [comment]
Ok, it seems like the "failure" message is just a response for not getting a response after some timeout.. it doesn't really indicate about something.

Also I've noticed that the eventer *do push* data.. but the heapster(metric) doesn't, although it say it does.

I can't really debug this out since it's not really clear for me how to build the heapster 😱 , but by using the code directly in a go script I wrote.. it seems like `elasticsearch.SaveDataIntoES` does work...

I'll appreciate some info about building :/

**Comment by AlmogBaku** [comment]
It seems like it push it without any special error from the library(i've putted some prints there), but I still can't see it in the ES

**Comment by AlmogBaku** [comment]
fixed by #1260

**Comment by piosz** [comment]
Thanks @AlmogBaku
